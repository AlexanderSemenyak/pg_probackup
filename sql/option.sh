#!/bin/sh

#============================================================================
# This is a test script for option test of pg_arman.
#============================================================================

BASE_PATH=`pwd`

# Clear environment variables used by pg_arman except $PGDATA.
# List of environment variables is defined in catalog.c.
unset BACKUP_PATH
unset ARCLOG_PATH
unset BACKUP_MODE
unset COMPRESS_DATA
unset KEEP_DATA_GENERATIONS
unset KEEP_DATA_DAYS

export PGDATA=$BASE_PATH/results/sample_database

# Note: not exported
BACKUP_PATH=$BASE_PATH/results/sample_backup2

# Setup backup catalog for backup test.
rm -rf $BACKUP_PATH
cp -rp data/sample_backup $BACKUP_PATH

# general option
pg_arman --help
pg_arman --version

# show option
# required argument check
pg_arman show
pg_arman show -B $BACKUP_PATH

# backup option
# required arguments check
pg_arman backup --verbose
pg_arman backup --verbose -B $BACKUP_PATH

# bad arguments check
pg_arman backup --verbose -B $BACKUP_PATH -b bad

# delete or validate requires DATE
pg_arman delete -B $BACKUP_PATH
pg_arman validate -B $BACKUP_PATH

# invalid configuration file check
echo " = INFINITE" > $BACKUP_PATH/pg_arman.ini
pg_arman backup --verbose -B $BACKUP_PATH
echo "BACKUP_MODE= " > $BACKUP_PATH/pg_arman.ini
pg_arman backup --verbose -B $BACKUP_PATH
echo "BACKUP_MODE=B" > $BACKUP_PATH/pg_arman.ini
pg_arman backup --verbose -B $BACKUP_PATH
echo "COMPRESS_DATA=FOO" > $BACKUP_PATH/pg_arman.ini
pg_arman backup --verbose -B $BACKUP_PATH
echo "TIMELINEID=-1" > $BACKUP_PATH/pg_arman.ini
pg_arman backup --verbose -B $BACKUP_PATH
echo "BACKUP_TARGETS=F" > $BACKUP_PATH/pg_arman.ini
pg_arman backup --verbose -B $BACKUP_PATH

# configuration priorityfile check
echo "BACKUP_MODE=ENV_PATH" > $BACKUP_PATH/pg_arman.ini
mkdir $BACKUP_PATH/conf_path
echo "BACKUP_PATH=$BACKUP_PATH/conf_path" > $BACKUP_PATH/pg_arman.conf
echo "BACKUP_MODE=CONF_PATH" > $BACKUP_PATH/conf_path/pg_arman.ini
mkdir $BACKUP_PATH/comm_path
echo "BACKUP_MODE=COMM_PATH" > $BACKUP_PATH/comm_path/pg_arman.ini
export BACKUP_PATH=$BACKUP_PATH
pg_arman backup --verbose
