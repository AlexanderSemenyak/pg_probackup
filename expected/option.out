\! sh sql/option.sh
pg_arman manage backup/recovery of PostgreSQL database.

Usage:
  pg_arman OPTION init
  pg_arman OPTION backup
  pg_arman OPTION restore
  pg_arman OPTION show [DATE]
  pg_arman OPTION validate [DATE]
  pg_arman OPTION delete DATE

Common Options:
  -D, --pgdata=PATH         location of the database storage area
  -A, --arclog-path=PATH    location of archive WAL storage area
  -B, --backup-path=PATH    location of the backup storage area
  -c, --check               show what would have been done
  -v, --verbose             output process information

Backup options:
  -b, --backup-mode=MODE    full or incremental
  -Z, --compress-data       compress data backup with zlib
  -C, --smooth-checkpoint   do smooth checkpoint before backup
  --validate                validate backup after taking it
  --keep-data-generations=N keep GENERATION of full data backup
  --keep-data-days=DAY      keep enough data backup to recover to DAY days age

Restore options:
  --recovery-target-time    time stamp up to which recovery will proceed
  --recovery-target-xid     transaction ID up to which recovery will proceed
  --recovery-target-inclusive whether we stop just after the recovery target
  --recovery-target-timeline  recovering into a particular timeline

Catalog options:
  -a, --show-all            show deleted backup too

Connection options:
  -d, --dbname=DBNAME       database to connect
  -h, --host=HOSTNAME       database server host or socket directory
  -p, --port=PORT           database server port
  -U, --username=USERNAME   user name to connect as
  -w, --no-password         never prompt for password
  -W, --password            force password prompt

Generic options:
  -q, --quiet               don't write any messages
  --debug                   debug mode
  --help                    show this help, then exit
  --version                 output version information, then exit

Read the website for details. <https://github.com/michaelpq/pg_arman>
Report bugs to <https://github.com/michaelpq/pg_arman/issues>.
pg_arman 0.1
ERROR: required parameter not specified: BACKUP_PATH (-B, --backup-path)
ERROR: required parameter not specified: ARCLOG_PATH (-A, --arclog-path)
ERROR: required parameter not specified: BACKUP_PATH (-B, --backup-path)
ERROR: Required parameter not specified: BACKUP_MODE (-b, --backup-mode)
ERROR: invalid backup-mode "bad"
ERROR: required delete range option not specified: delete DATE
INFO: validate: 2009-05-31 17:05:53 backup and archive log files by CRC
INFO: validate: 2009-06-01 17:05:53 backup and archive log files by CRC
WARNING: CRC of backup file "PG_VERSION" must be 0 but FEF71BC1
WARNING: backup 2009-06-01 17:05:53 is corrupted
WARNING: syntax error in " = INFINITE"
ERROR: Required parameter not specified: BACKUP_MODE (-b, --backup-mode)
ERROR: invalid backup-mode ""
ERROR: invalid backup-mode "B"
ERROR: option -Z, --compress-data should be a boolean: 'FOO'
ERROR: invalid option "TIMELINEID"
ERROR: invalid option "BACKUP_TARGETS"
ERROR: invalid backup-mode "ENV_PATH"
